lib/std/syncio.nim

```diff
@@ -689,7 +691,7 @@ when defined(posix) and not defined(nimscript):
 
 proc open*(f: var File, filename: string,
           mode: FileMode = fmRead,
-          bufSize: int = -1): bool {.tags: [], raises: [], benign.} =
+          bufSize: int = -1): bool {.tags: [], raises: [IOError], benign.} =
```

Comment:
adding new exceptions is a breaking change (because it changes the meaning of programs that catch specific exceptions) - this should be done via a new symbol